src/Symfony/Component/Yaml/Inline.php

```diff
@@ -513,16 +513,16 @@ private static function parseStructure(StringReader $reader, $closingMarker, cal
             $parseItem();
 
             $reader->consumeWhiteSpace();
-            $separator = $reader->readChar(',');
-            $reader->consumeWhiteSpace();
-            if (!$separator) {
+            if (!$reader->readChar(',')) {
                 $reader->expectChar($closingMarker);
                 break;
-            } else {
-                // bc support of [ foo, , ]
-                while ($reader->readChar(',')) {
-                    $reader->consumeWhiteSpace();
-                }
+            }
+
+            $reader->consumeWhiteSpace();
+
+            // bc support of [ foo, , ]
+            while ($reader->readChar(',')) {
+                $reader->consumeWhiteSpace();
```

Comment:
This section looks like one of those rare cases where a `do {} while()` makes sense.